### PR TITLE
[Modernization] Replace Internal Backticks with Shell Expansion: Rscript 

### LINKED ISSUE #6523: [Modernization] Replace Internal Backticks with Shell Expansion
## Goal
Modernize the rsyslog test suite by replacing legacy internal backtick expansion (e.g., ``file=`echo $RSYSLOG_OUT_LOG` ``) with standard shell expansion (e.g., `file="'$RSYSLOG_OUT_LOG'"`). 

**Why?** 
The legacy pattern forces rsyslog to fork a shell at runtime for every expansion. Modern shell expansion is more efficient, faster, and follows current project best practices.

## For Contributors (AI-Assisted Workflow)
This issue is designed to be completed using an AI agent (like Gemini CLI, Aider, Cursor, or ChatGPT with codebase access). 

**Instructions:**
1. Use the **Mission Prompt** below to guide your AI agent.
2. **Limit your work to ONE module or batch (max 10-15 files)** per Pull Request to ensure a smooth review process.
3. Check off the module you are working on in the "Progress Tracking" section below.

---

### 🤖 The Mission Prompt
Copy and paste this prompt into your AI agent:

> "Mission: Modernize rsyslog test scripts by replacing internal backticks with shell expansion.
>
> 1. **Discovery & Selection**: 
>    - Search the `tests/` directory for `.sh` files containing the pattern '`echo $'. 
>    - Pick a specific category or module (e.g., 'imfile', 'imuxsock', 'rscript', etc.).
>    - **Limit this session to a maximum of 10-15 files** to keep the PR manageable.
>    - List the files you found and ask me: 'I suggest starting with these X files. Should I proceed?'
>
> 2. **Filtering (Safety)**: 
>    - **EXCLUDE** any file where the filename contains 'backtick' (e.g., `rscript_backticks.sh`).
>    - **EXCLUDE** any file where comments indicate it specifically tests the 'backtick feature'.
>    - *Reason: We must keep actual feature tests in their original format.*
>
> 3. **Transformation**: 
>    - For the selected files, update the `add_conf` blocks.
>    - Replace: `file=`echo $VAR``  ==>  `file="'$VAR'"`
>    - Replace: `binary=`echo $VAR`` ==> `binary="'$VAR'"`
>    - *Note: The quoting format "'$VAR'" is critical so the bash shell expands the variable before passing it to rsyslog.*
>
> 4. **Verification**: 
>    - For every modified file, run its specific test: `make -C tests [FILENAME].log`
>    - Ensure you see a 'PASS' result. Do not commit files that fail.
>
> 5. **Finalize**: 
>    - Show me the diff for the changes.
>    - Summarize the work (e.g., 'Modernized 10 imuxsock tests')."

---

## Progress Tracking
Please comment below to claim a batch. **Limit: 1 batch per PR.**

### Categories / Batches
- [ ] **imuxsock** tests (~15 files)
- [ ] **imfile** Batch 1 (~15 files)
- [ ] **imfile** Batch 2 (~15 files)
- [ ] **imtcp / imudp** tests (~20 files)
- [ ] **rscript** tests Batch 1 (~15 files)
- [ ] **rscript** tests Batch 2 (~15 files)
- [ ] **mmanon / mmnormalize** tests (~15 files)
- [ ] **Miscellaneous** (Any other module)

## Technical Reference
- **Legacy Pattern (Inside add_conf):** `action(type="omfile" file=`echo $RSYSLOG_OUT_LOG`)`
- **Modern Pattern (Inside add_conf):** `action(type="omfile" file="'$RSYSLOG_OUT_LOG'")`

### PR DESCRIPTION
**Pull request by aditiprakash**
[Modernization] Replace Internal Backticks with Shell Expansion: Rscript 
This modernizes the rscript test suite by replacing legacy backtick-based file expansions
(e.g. file=echo $RSYSLOG_OUT_LOG) with explicit quoted shell variable expansion
(e.g. file="'$RSYSLOG_OUT_LOG'"). This improves maintainability and aligns with
current shell best practices for CI and container environments.

Before: tests relied on backtick command substitution with inconsistent quoting
and limited nesting semantics.
After: tests pass the expanded filename via a quoted shell variable literal,
making the expansion semantics explicit and more predictable.

Impact: test execution semantics may change slightly where previous backtick/quoting
behavior masked word-splitting or nesting edge cases. No rsyslog runtime logic,
message flow, or module behavior is modified — the updates are limited to test
scripts (12 rscript tests updated). This change reduces ambiguity and improves
reproducibility across environments.

Fixes: https://github.com/rsyslog/rsyslog/issues/6523

### CODE REVIEW

**Comment by aditiprakash** [comment]
The suggested fix has been applied, not sure why the daily runs are failing

File: tests/rscript_re_extract.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ input(type="imtcp" port="0" listenPortFileName="'$RSYSLOG_DYNNAME'.tcpflood_port
 add_conf "
 set \$.number = re_extract(\$msg, '.* ([0-9]+)$', 0, 1, 'none');"
 add_conf '
-action(type="omfile" file=`echo $RSYSLOG_OUT_LOG` template="outfmt")
+'action(type="omfile" file="'$RSYSLOG_OUT_LOG'" template="outfmt")
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/rscript_re_extract.sh]: Addressed in 36c6392332475e2fa53e885e7677342e59711bde: the stray leading quote in tests/rscript_re_extract.sh was removed, leaving the intended file="''" expansion inside the generated rsyslog config.

File: tests/rscript_hash32.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ input(type="imtcp" port="0" listenPortFileName="'$RSYSLOG_DYNNAME'.tcpflood_port
 set $.hash_no_1 = hash32("0f9a1d07-a8c9-43a7-a6f7-198dca3d932e");
 set $.hash_no_2 = hash32mod("0f9a1d07-a8c9-43a7-a6f7-198dca3d932e", 100);
 
-action(type="omfile" file=`echo $RSYSLOG_OUT_LOG` template="outfmt")
+'action(type="omfile" file="'$RSYSLOG_OUT_LOG'" template="outfmt")
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/rscript_hash32.sh]: Addressed in 36c6392332475e2fa53e885e7677342e59711bde: the stray leading quote in tests/rscript_hash32.sh was removed, leaving the intended file="''" expansion inside the generated rsyslog config.

[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/rscript_hash32.sh]: Clarification: the intended generated config now uses the RSYSLOG_OUT_LOG output path correctly; my previous reply had its inline variable snippet mangled while posting. The underlying stray leading quote in tests/rscript_hash32.sh is fixed in 36c6392332475e2fa53e885e7677342e59711bde.

File: tests/rscript_re_extract.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/rscript_re_extract.sh]: Clarification: the intended generated config now uses the RSYSLOG_OUT_LOG output path correctly; my previous reply had its inline variable snippet mangled while posting. The underlying stray leading quote in tests/rscript_re_extract.sh is fixed in 36c6392332475e2fa53e885e7677342e59711bde.

**Review by rgerhards — APPROVED** [review]
looks like the unrelated stuff was a flake
